src/mflux_debugger/profiler_service.py

```
@@ -0,0 +1,281 @@
+        return {
+            "total_calls": len(self.profile.calls),
+            "unique_functions": len(unique_functions),
+            "duration": duration,
+            "max_depth_reached": max((c.depth for c in self.profile.calls), default=0),
+            "top_functions": sorted_functions[:20],  # Top 20 most called
+        }
+
+    def save(self, output_dir: Path | None = None) -> Path:
+        """Save profile to JSON file."""
+        if not self.profile:
+            raise RuntimeError("No profile data to save")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)